### DISCUSSION TITLE
TypeError: catching classes that do not inherit from BaseException is not allowed

### DISCUSSION THREAD

**Discussion by GerBreOwn** [sqlalchemy/sqlalchemy#8552] [Usage Questions]
I am new to sqlalchemy and I am getting the above error.
I had Sqlalchemy 1.4.31 and upgraded to 1.4.41 or whatever the latest version is. I am getting the error with both versions.
I am running Python3.10.5 on a Raspberry Pi 4 - 8 gb machine with Ubuntu 22.04 OS.
Is there any other information you might need to help solve this problem?
Thanks

**Comment by zzzeek** [comment]
hi -

youd want to make sure you are writing [try / except](https://pythonbasics.org/try-except/) blocks correctly.    This is not really a SQLAlchemy thing.

**Reply by GerBreOwn** [reply]
I do NOT have any try/except blocks in my code. 
All of the errors are in the Sqlalchemy code at sqlalchemy/pool/base.py has 4 errors at lines 498, 790, 533, 329 and 1 error in sqlalchemy/pool/impl.py line 113.
Any other suggestions to fix this problem as my application is at a stand still until the problem is fixed.
Thanks for the suggestion.

**Reply by zzzeek** [reply]
ok -

can you please share the exact SQLAlchemy version in use as well as a complete stack trace?

**Reply by CaselIT** [reply]
At line `sqlalchemy/pool/impl.py line 113.` there is this except:
https://github.com/sqlalchemy/sqlalchemy/blob/bbd90a987eea70c28dd1e2ee7007722b16ec9f74/lib/sqlalchemy/pool/impl.py#L113
that is definitely an exception:
https://github.com/sqlalchemy/sqlalchemy/blob/bbd90a987eea70c28dd1e2ee7007722b16ec9f74/lib/sqlalchemy/util/queue.py#L41-L44

so the issue here seems something in your local install. I would suggest uninstalling and re-installing sqlalchemy

**Reply by zzzeek** [reply]
it looks like installation but also something in the module space could be overwriting util/queue.py.

I would debug this by pdbing in and seeing what sqla_queue.Full is, `repr(sqla_queue.Full)`, `sqla_queue.Full.__module__`, things like that

**Reply by GerBreOwn** [reply]
I had a earlier version that had the same problem. I then upgraded to the latest version 1.4.41 and the problem is still there. Could it be possible that because I am running it on a Raspberry Pi it is causing the problem. Other Python programs do not have that problem. I was also thinking what version of Python is used to write SQLAlchemy? I am running Python 3.10.5 on my system and if SQLAlchemy is developed with an older version of Python something might have changed between Python versions.
Thanks for all of the replies to this problem.

**Reply by GerBreOwn** [reply]
I just copied my app from the Raspberry Pi to my laptop. When I run it there I get the following error: "File "/home/gerald/.local/lib/python3.10/site-packages/sqlalchemy/dialects/postgresql/psycopg2.py", line 811, in dbapi
    import psycopg2
ModuleNotFoundError: No module named 'psycopg2'"
The reason for this error is that I have psycopg3 or psycopg installed as that is the latest version of psycopg available. It is now up to 3.1. It will not let me install psycopg2 with a lot of errors when I tried that.
I am also running this version on the RPi.
This is getting weirder and weirder.

**Reply by zzzeek** [reply]
OK, "copying", what's that?   are you using `pip` to install libraries or just copying files into site-packages?

**Reply by GerBreOwn** [reply]
I copied all of my app code from from one computer to another. I then had to PIP install Pandas, SQLAlchemy and some other libraries on the second computer. It then would not run because SQLAlchemy was trying to import psycopg2 and I have psycopg3 installed, which was just upgraded to 3.1.

My setup further information: I import Pandas and SQLAlchemy, make the connections, df = pd.read_sql(sql, conn), print df, which prints my dataframe(df), conn.close(). After these steps the SQLAlchemy errors disappear, but the program also stops. I have additional code that is supposed to be working with the Panda data to create some charts.

**Reply by zzzeek** [reply]
if your own application has modules named "queue" or something like that, although that's more of an old Python 2 problem.   I think pdb'ing into the error is the most direct way to see what it's thinking.    re: raspberry pi I have a few here and I dont know that there's any special thing happening on them that would cause imports to be screwy.

**Reply by CaselIT** [reply]
This seems like an environment issue of your side.

My advice would be to try creating a new python environment, install just sqlalchemy, pandas and psycopg3  and try to connect and read a table to a dataframe. If that work try installing all other requirement until the code breaks.

**Reply by zzzeek** [reply]
how are we using psycopg3 if this user is using released 1.4.x SQLAlchemy versions?

**Reply by CaselIT** [reply]
right. that would seem strange indeed

**Reply by GerBreOwn** [reply]
Sorry for the late reply but I have been sick in bed for the past few days. Tonight I created an environment and did NOT get any errors in the env NOR in my original program. Maybe my sickness was able to flush the problem out of the computer system as well as from my system.
Thanks for all of your suggestions and hopefully the problem has been solved/fixed so I can continue on with my application.
